hermeto/core/package_managers/javascript/pnpm/resolver.py

```
@@ -199,19 +202,17 @@ def _create_workspace_components(
+def _load_pnpm_workspace(project_dir: Path) -> dict[str, Any]:
+    pnpm_workspace_path = project_dir / "pnpm-workspace.yaml"
     if not pnpm_workspace_path.exists():
-        return []
+        return {}
 
     try:
         with pnpm_workspace_path.open() as f:
-            pnpm_workspace = yaml.safe_load(f)
+            return yaml.safe_load(f) or {}
     except yaml.YAMLError:
         raise PackageRejected(f"The {pnpm_workspace_path} file must contain valid YAML.")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)